frontend/test/metabase/scenarios/models/model-actions.cy.spec.js

```diff
@@ -0,0 +1,201 @@
+    fieldSettings()
+      .findByText(/Number/i)
+      .click();
+    cy.findByRole("button", { name: "Save" }).click();
+    modal().within(() => {
+      cy.findByLabelText("Name").type("Delete Order");
+      cy.findByRole("button", { name: "Create" }).click();
+    });
+    cy.findByLabelText("Action list")
+      .findByText("Delete Order")
+      .should("be.visible");
+
+    openActionEditorFor("Delete Order");
+    fillQuery(" AND status = 'pending'");
+    fieldSettings()
+      .findByRole("radiogroup", { name: /Field type/i })
+      .findByLabelText(/Number/i)
+      .should("be.checked");
+    cy.findByRole("button", { name: "Update" }).click();
+
+    cy.findByLabelText("Action list")
+      .findByText(
+        "DELETE FROM orders WHERE id = {{ id }} AND status = 'pending'",
+      )
+      .should("be.visible");
+  });
+
+  it("should allow to make actions public and execute them", () => {
+    cy.intercept("/api/public/action/*/execute", request => {
+      expect(request.body).to.deep.equal({
+        parameters: { [TEST_PARAMETER.id]: -2 },
+      });
+    });
+
+    cy.get("@modelId").then(modelId => {
+      cy.request("POST", "/api/action", {
+        ...SAMPLE_QUERY_ACTION,
+        model_id: modelId,
+      });
+      cy.visit(`/model/${modelId}/detail`);
+      cy.wait("@getModel");
+    });
+
+    cy.findByText("Actions").click();
+    openActionEditorFor(SAMPLE_QUERY_ACTION.name);
+
+    cy.findByRole("dialog").within(() => {
+      cy.findByRole("button", { name: "Action settings" }).click();
+      cy.findByLabelText("Make public").should("not.be.checked").click();
+      cy.findByLabelText("Public action link URL")
+        .invoke("val")
+        .then(url => {
+          cy.wrap(url).as("publicUrl");
+        });
+    });
+
+    cy.get("@publicUrl").then(url => {
+      cy.signOut();
+      cy.visit(url);
+    });
+
+    cy.findByLabelText(TEST_PARAMETER.name).type("-2");
+    cy.findByRole("button", { name: "Submit" }).click();
+    cy.findByText("Thanks for your submission.").should("be.visible");
+    cy.findByRole("form").should("not.exist");
+    cy.findByRole("button", { name: "Submit" }).should("not.exist");
+
+    cy.signInAsAdmin();
+    cy.get("@modelId").then(modelId => {
+      cy.visit(`/model/${modelId}/detail`);
+      cy.wait("@getModel");
+    });
+
+    cy.findByText("Actions").click();
+    openActionEditorFor(SAMPLE_QUERY_ACTION.name);
+
+    cy.findByRole("dialog").within(() => {
+      cy.findByRole("button", { name: "Action settings" }).click();
+      cy.findByLabelText("Make public").should("be.checked").click();
+    });
+    modal().findByRole("button", { name: "Yes" }).click();
+
+    cy.get("@publicUrl").then(url => {
+      cy.visit(url);
+      cy.findByRole("form").should("not.exist");
+      cy.findByRole("button", { name: "Submit" }).should("not.exist");
+      cy.findByText("An error occurred.").should("be.visible");
+    });
```

Comment:
I don't think this is actually testing that we can run actions from public actions pages? I think we should omit this bit for now until we can really test a working execution.

I'm sorry it's been slow, but I should have actions setup/reset ready to use next week.
https://github.com/metabase/metabase/pull/27871

Reply:
You're right, although we're testing a bunch of different things here:

* enabling public sharing
* opening a public action page while not being signed in
* submitting the action form
* disabling public sharing + ensuring we won't be able to access the page anymore

We test execution up to the point where the page sends a `POST /execute` request. It doesn't count as a complete E2E for sure, but IMO we'd end up in a more reliable state if we merge this and then later patch the test to ensure our action actually mutates the database. What do you think?

No worries about the setup/reset tooling, I really appreciate your help here!

Reply:
oh yeah, I think the test is valuable, I didn't mean we should remove the whole thing, just the part where we intentionally throw an error.  maybe just update the description to remove "and execute them"

Reply:
We actually execute an action, it just doesn't affect data (see the "Action query" section in the description).
The highlighted block verifies that the public action page becomes unavailable as soon as public sharing is turned off